torch/fx/passes/dialect/aten/canonicalize.py

```diff
@@ -0,0 +1,43 @@
+from torch.fx.graph_module import GraphModule
+from torch._ops import OpOverload, OpOverloadPacket
+
+def is_canonical(gm: GraphModule) -> bool:
+
+    for node in gm.graph.nodes:
+        if node.op in {"placeholder", "output"}:
+            continue
+
+        if node.op == "call_method":
+            # canonical aten graph couldn't have call_method nodes
+            return False
+
+        if node.op == "get_attr":
+            return False
+
+        if node.op == "call_module":
+            continue
```

Comment:
Are all module calls considered "canonical"?

I guess I'm wondering if the main purpose of modules being in the graph right now is for opaque / custom fused ops. What about if we see a bunch of `torch.nn` modules in the graph?

Reply:
AOTAutograd would trace everything to the op-level nodes, so if we see call_module nodes in graph, I suppose it should be opaque module fused by some pass. 
I would expect it to be GraphModule, but not nn.Module.